Fake user data capability extension

5queezer opened the pull request:
Fake user data extension. Ability to create and delete fake users. Fake posts can be assigned to a fake user through a variable. See example in the Readme.

Commit:
User data generation added

Commit:
readme extended

Commit:
package domain

5queezer committed:
readme, composer package name

5queezer committed:
readme

haroldangenent commented:
First off, thanks for you contribution to this repo! I didn't get a chance to look at any real logic yet, but I've noticed a few things in this PR at a glance:

- The tests for existing functions seem to fail. All tests should still pass;
- Your newly added features do not seem to include tests. However, we would like all newly added code to have automated tests;
- The package name and homepage of the Composer package seem to be changed. Also, there seems to be a dependency added to the NPM packages and some other unrelated changes to this PR. Could you revert changes to these meta files?
- The coding standard PSR is used for this codebase. I think there's a gulpfile with a tasks that checks the coding standards for you. Could you make sure your changes conform to PSR coding standards?

I think faking users could be a useful feature, thanks again.

haroldangenent reviewed (changes requested):
See review comments above.

5queezer committed:
Revert "package domain"

This reverts commit c989a51bd0ee1617938f02d131f699551005a72f.

5queezer committed:
Revert "readme, composer package name"

This reverts commit d23bfe276e15f012ac73b7303d4c5f2ada045a69.

5queezer committed:
readme content error

5queezer committed:
revert package.json to original state

5queezer committed:
added package-lock and phpstorm ide folder to gitignore

5queezer committed:
newline at end of file

5queezer committed:
5quuezer reverted to trendwerk

5queezer committed:
Command.php whitespace reverted

5queezer committed:
additional author removed

5queezer committed:
string 'posts' changed to 'objects'

5queezer committed:
cucumber test feature user

5queezer committed:
datatypes removed from comments

5queezer commented:
Hello Harold,

- All tests pass.
- I have included a test feature for _users_. However, I could not test if the posts are assigned to a user. I hope that's good for now.
- I have reverted the composer.json and package.json to their previous state. I also removed the author section in composer.json.
- PSR check was successful

I hope you give the green light!

haroldangenent commented on .gitignore:
```diff
@@ -2,3 +2,5 @@ composer.lock
 node_modules
 vendor
+package-lock.json
+.idea/
```
Could you remove `.idea/` from this file? Since it's editor-based, I'd assume this in `.gitignore_global`.

5queezer replied:
`.idea/` and `package-lock.json` have been removed. `package-lock.json` has not been pushed to the repo.

haroldangenent commented on README.md:
```diff
@@ -133,6 +136,22 @@ Trendwerk\Faker\Entity\Post:
 
 Currently the only type of supported attachments are images.
 
+### Users
```
The placement of this part is strange, because it breaks up the "Attachments" area. Maybe `Users` could be above the `Attachments` heading?

5queezer replied:
The _User_ heading has been placed above the _Attachments_ heading.

haroldangenent commented on composer.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "trendwerk/faker",
-  "description": "Fake post data with WP-CLI.",
+  "description": "Fake post and user data with WP-CLI.",
```
Maybe just `Fake data with WP-CLI.`?

5queezer replied:
Agreed.

haroldangenent commented on init.php:
```diff
@@ -1,12 +1,13 @@
 <?php
```
Could you remove coding standard preferences from this file, to keep the changes as clean as possible?

5queezer replied:
Sorry, I was not aware of this. My IDE does this. The file was reverted to its previous state.

haroldangenent commented on init.php:
```diff
@@ -1,12 +1,13 @@
-WP_CLI::add_command('faker', __NAMESPACE__ . '\\Command', [
-    'shortdesc' => 'Fake post data with WP-CLI.',
+WP_CLI::add_command('faker', __NAMESPACE__.'\\Command', [
+    'shortdesc' => 'Fake post and user data with WP-CLI.',
```
Maybe also `Fake data with WP-CLI.`?

5queezer replied:
It has been modified as well. Sorry for the whitespace modifications. My IDE was doing this and I was not aware of it.

haroldangenent commented on src/Entity/User.php:
```diff
@@ -0,0 +1,73 @@
+
+class User extends Entity
+{
+    public $user_pass; // The plain-text user password.
```
Could you remove comments from user-properties (they are documented at `wp_insert_user`, which is already in the README).

5queezer replied:
The comments have been removed.

haroldangenent commented on features/fake-user.feature:
```diff
@@ -0,0 +1,66 @@
+        10
+        """
+    
+  Scenario: Generate posts with an user
```
As you've stated, it's a bit harder to test for post-count by a certain number of users. However, the test below doesn't really test anything other that the other test (post creation is already tested in `fake-post.feature`).

I'm either for removing the test, or actually testing if the fake users actually own 15 posts. Maybe something like this could work?

```
wp post list --author=$(wp user list --meta_key=_fake --format=json --field=ID) --format=count
```

5queezer replied:
I tried this but I could not have success. Obviously there is no `--author` parameter, or if it does, it does not process a json array, but a single user id.

I removed that part entirely.

haroldangenent reviewed (changes requested):
Sorry for my late response, unfortunately I have been out sick last week. The PR looks good, thanks for your work on this!

I just have some very minor issues to keep the codebase consistent, but everything seems to work like a charm. The user-post connection test could also be improved, I've added an example on what might work.

Commit:
init.php reverted to old coding standards

Commit:
short description modified

Commit:
comments removed

Commit:
post generation removed

Commit:
package-lock.json removed from gitignore

Commit:
new short description

Commit:
user has been placed above the attachment heading

haroldangenent reviewed (approved):
@5queezer This is looking great, thanks for your contribution! I'll create a new release for this, since I'm not planning any maintenance releases in the near future.

5queezer commented:
Thank you for helping me and pointing out the errors!
